automation/Extension/firefox/data/content.js

```diff
@@ -351,106 +354,180 @@ function getPageScript() {
+        try {
+          instrumentObjectProperty(object, objectName, properties[i], logSettings);
+        } catch(error) {
+          logErrorToConsole(error);
         }
-      } catch(err) {
-        // can't access static properties on prototypes
-        logProperty(object, objectName, propertyName, logSettings);
       }
     }
+    if (testing) {
+      window.instrumentObject = instrumentObject;
+    }
 
-    // Log calls to object/prototype methods
-    function logFunction(object, objectName, method, logSettings) {
-      var originalMethod = object[method];
-      object[method] = function () {
+    // Log calls to a given function
+    // This helper function returns a wrapper around `func` which logs calls
+    // to `func`. `objectName` and `methodName` are used strictly to identify
+    // which object method `func` is coming from in the logs
+    function instrumentFunction(objectName, methodName, func, logSettings) {
+      return function () {
         var callContext = getOriginatingScriptContext(!!logSettings.logCallStack);
-        logCall(objectName + '.' + method, arguments, callContext, logSettings);
-        return originalMethod.apply(this, arguments);
+        logCall(objectName + '.' + methodName, arguments, callContext, logSettings);
+        return func.apply(this, arguments);
       };
     }
 
     // Log properties of prototypes and objects
-    function logProperty(object, objectName, property, logSettings) {
-      var propDesc = Object.getPropertyDescriptor(object, property);
+    function instrumentObjectProperty(object, objectName, propertyName, logSettings={}) {
+
+      // Store original descriptor in closure
+      var propDesc = Object.getPropertyDescriptor(object, propertyName);
       if (!propDesc){
-        console.log("logProperty error", objectName, property, object);
+        console.error("Property descriptor not found for", objectName, propertyName, object);
         return;
       }
-      // store the original getter in the closure
+
+      // Instrument data or accessor property descriptors
       var originalGetter = propDesc.get;
       var originalSetter = propDesc.set;
-      // TODO what if the getter is undefined
-      Object.defineProperty(object, property, {
+      var originalValue = propDesc.value;
+
+      // We overwrite both data and accessor properties as an instrumented
+      // accessor property
+      Object.defineProperty(object, propertyName, {
         configurable: true,
         get: (function() {
-          return function(){
-            if (!originalGetter){
-              console.log("originalGetter is undefined!")
-              logValue(objectName + '.' + property, "", "get(failed)", callContext, logSettings);
+          return function() {
+            var origProperty;
+            var callContext = getOriginatingScriptContext(!!logSettings.logCallStack);
+
+            // get original value
+            if (!originalGetter && !('value' in propDesc)) {
```

Comment:
Any reason to use negations in this if-else clause?
I though could be written more "positively", by something like `if (originalValue) else if (originalGetter) else { error } ...`